src/main/java/picocli/CommandLine.java

```diff
@@ -7081,7 +7090,21 @@ public String qualifiedName(String separator) {
              * @return this CommandSpec for method chaining
              * @since 3.1 */
             public CommandSpec aliases(String... aliases) {
+                Set<String> existingAliasSet = this.aliases;
                 this.aliases = new LinkedHashSet<String>(Arrays.asList(aliases == null ? new String[0] : aliases));
+                if (parent != null) {
+                    //remove & add aliases
+                    Set<String> newAliasSet = new LinkedHashSet<String>(this.aliases);
```

Comment:
Another minor detail: I would call this variable `addedAliasSet` since the intention is to subtract the old set so that only the _added_ alias remain, and then perform some operation on only the _added_ aliases.

Reply:
Done